system/jlib/jexcept.cpp

```diff
@@ -1452,6 +1452,44 @@ void printStackReport(__int64 startIP)
 
 //---------------------------------------------------------------------------------------------------------------------
 
+bool getAllStacks(StringBuffer &output)
+{
+#ifdef __linux__
+    StringBuffer exePath;
+    char *_exePath = exePath.reserveTruncate(NAME_MAX + PATH_MAX +1);
+    ssize_t len = readlink("/proc/self/exe", _exePath, NAME_MAX + PATH_MAX +1);
+    if (-1 == len)
+    {
+        output.append("Unable to capture stacks");
+        return false;
+    }
+    _exePath[len] = '\0';
+    VStringBuffer cmd("gdb --batch -n -ex 'thread apply all bt' %s %u", _exePath, GetCurrentProcessId());
+    Owned<IPipeProcess> pipe = createPipeProcess();
+    if (pipe->run("get stacks", cmd, nullptr, false, true, false))
+    {
+        Owned<ISimpleReadStream> pipeReader = pipe->getOutputStream();
+        const size32_t chunkSize = 8192;
```

Comment:
This block of code should be in a common jlib function - I can find at least 5 copies of the same code! (Function should read a string from a ISimpleReadStream.)

Reply:
Agree, I've commoned up in the latest commit